map/src/map/layers/LocalClientTrackLayer.js

```diff
@@ -67,6 +67,20 @@ export default function LocalClientTrackLayer() {
                 checkUpdateLayers();
             }
         }
+        if (ctx.selectedGpxFile?.layers) {
+            /*
+                map.off('click');
+                map.on('click', clickMap);
+
+                clickMap() uses context vars?
+                Got old context inside event handler?
+
+                1. Use Effect() to monitor context changes.
+                2. Disable previous handler (don't queue handlers).
+                3. Re-setup new event handler with refreshed context.
+             */
+            addClickOnMap(); // refresh ctx
```

Comment:
Need to fix getTrackWithAnalysis - replace ctx with current ctx.selectedGpxFile (trackRef.current for createNewRouteLine)

Reply:
dealt another way